### ISSUE TITLE
Optional/toggle-able sections in workflows

### ISSUE THREAD

**Issue by stodoran** [warpdotdev/Warp#5465]
### Dupe Check

- [X] I have [searched Warp feature requests](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3A%22Feature+requests%22) and there are no duplicates

### Describe the solution you'd like?

My feature request is to be able to make part of the text in a workflow optional. How this would work:

Example workflow is below. Any text inside double square braces is optional. These portions of the workflow are tabbed to just like arguments, and can be toggled in a similar way to an enum. If toggled off, the text could be dimmed to show it won't be included when the command is run, or it could be cut out and replaced sort of icon indicating the cut such as a notch (which can be tabbed to again to re-enabled the section).
```
foo [[--bar]]
```

A toggle-able section should be able to wrap arguments. For example, in the below workflow the count argument only gets tabbed to if the user first sets the section it resides in to enabled when first tabbing through the command.
```
foo [[--bar {{count}}]]
```

### Is your feature request related to a problem? Please describe.

This would primarily be useful for flags/arguments to commands called within workflows, or to make entire lines/sections of a workflow be toggle-able.

Right now there is a workaround if I have a workflow with a boolean flag that should sometimes be passed but not always. I can make a flag optional by creating an enum with two values:
* A single space (don't provide the flag)
* The flag value

Here is a screenshot of a minimal example:
![image](https://github.com/user-attachments/assets/c3794b30-cd3e-46fd-9b5f-906958899e72)
Notice how the "bar" flag is made optional by have a single space as the first option in the enum, and the flag value itself as the second option in the enum.

This works for creating an optional flag, but has the following drawbacks:
* Feels like a hack
* Takes longer to set up than the feature I've described would
* In a command with many optional flags, the command can end up with many spaces in a row (not readable)
* Cannot nest an argument within (biggest drawback, makes the enum workaround only viable for boolean flags)

The use of the proposed feature would also extend beyond flags, for example if the user wishes to conditionally run some lines of the workflow the current best solution is to use an if statement and make the condition an enum. With this feature proposal those lines could just be wrapped in braces.

In short: this proposed feature would make it easier to create a workflow with optional flags and cleaner to use such a workflow. It would allow optional use of workflow arguments, and it would make calling certain lines of the workflow conditionally more simple.

### How important is this feature to you?

3

### Warp Internal (ignore) - linear-label:39cc6478-1249-4ee7-950b-c428edfeecd1

None

**Comment by stodoran** [comment]
Another improvement for workflows would be if an enum's values could come from the output of some set of commands, or from an environment variable. For example a simple workflow one might create with this would be one which activates a python venv, and the {{venv}} variable in the workflow would be an enum whose values come from `ls /foo/venvs/`.
